data/helpers.d/utils

```diff
@@ -24,3 +24,64 @@ ynh_get_plain_key() {
         fi
     done
 }
+
+# Restore a previous backup if the upgrade process failed
+#
+# usage:
+# ynh_backup_before_upgrade
+# ynh_clean_setup () {
+# 	ynh_backup_after_failed_upgrade
+# }
+# ynh_abort_if_errors
+#
+ynh_backup_after_failed_upgrade () {
+	echo "Upgrade failed." >&2
+	app_bck=${app//_/-}	# Replace all '_' by '-'
+	
+        # Check if an existing backup can be found before removing and restoring the application.
+	if sudo yunohost backup list | grep -q $app_bck-pre-upgrade$backup_number
+    	then
+		# Remove the application then restore it
+		sudo yunohost app remove $app
+		# Restore the backup
+		sudo yunohost backup restore --ignore-hooks $app_bck-pre-upgrade$backup_number --apps $app --force
```

Comment:
`--ignore-hooks` should now be replaced by `--ignore-system` (see [here](https://github.com/YunoHost/yunohost/blob/unstable/src/yunohost/backup.py#L1965))